misk/src/main/kotlin/misk/logging/LogRateLimiter.kt

```diff
@@ -0,0 +1,56 @@
+package misk.logging
+
+import misk.web.interceptors.LogRequestResponse
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Holds rate limiters for logging success and error responses. There is a rate limiter for every
+ * action and service that calls it. The rate limiters are created according to the value set in
+ * [LogRequestResponse] annotation, which is the number of buckets per pod per service.
+ */
+@Singleton
+class LogRateLimiter @Inject constructor(
+  private val rateLimiterFactory: RateLimiter.Factory
+) {
+  private val liveConfigRef =
+    AtomicReference(LiveConfig(
+      rateLimiters = ConcurrentHashMap<LogBucketId, RateLimiter>(),
+      errorRateLimiters = ConcurrentHashMap<LogBucketId, RateLimiter>()
```

Comment:
Why not just make `error` a boolean field on the `LogBucketId`? Then you can just have a single ConcurrentHashMap. And you just need a single `tryAcquire` method

Reply:
Good point, adding.